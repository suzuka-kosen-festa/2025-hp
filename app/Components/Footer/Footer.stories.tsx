import type { Meta, StoryObj } from "@storybook/react";
import { createRemixStub } from "@remix-run/testing";
import FooterComponent from "./Footer";

const Footer = FooterComponent;

const meta: Meta<typeof Footer> = {
  title: "Footer",
  component: Footer,
  decorators: [
    (story) => {
      const remixStub = createRemixStub([
        {
          Component: () => story(),
          path: "/*",
        },
      ]);

      return remixStub({});
    },
  ],
  parameters: {
    layout: "fullscreen",
  },
  tags: ["autodocs"],
  argTypes: {
    siteMap: {
      description: "サイトマップのセクションとリンクの配列",
    },
  },
};

export default meta;

type Story = StoryObj<typeof Footer>;

const extensiveSiteMap = [
  {
    sectionTitle: "ホーム",
    links: [
      { title: "トップページ", href: "/" },
      { title: "お知らせ", href: "/news" },
      { title: "イベント情報", href: "/events" },
      { title: "アクセス", href: "/access" },
    ],
  },
  {
    sectionTitle: "高専祭について",
    links: [
      { title: "高専祭とは", href: "/about" },
      { title: "実行委員会", href: "/committee" },
      { title: "歴史", href: "/history" },
      { title: "理念・目標", href: "/philosophy" },
    ],
  },
  {
    sectionTitle: "参加・出展",
    links: [
      { title: "一般参加", href: "/participation" },
      { title: "出展申込", href: "/exhibition" },
      { title: "スケジュール", href: "/schedule" },
      { title: "会場マップ", href: "/map" },
      { title: "注意事項", href: "/notes" },
    ],
  },
  {
    sectionTitle: "サポート",
    links: [
      { title: "よくある質問", href: "/faq" },
      { title: "お問い合わせ", href: "/contact" },
      { title: "プライバシーポリシー", href: "/privacy" },
    ],
  },
  {
    sectionTitle: "アーカイブ",
    links: [
      { title: "過去の高専祭", href: "/archive" },
      { title: "写真ギャラリー", href: "/gallery" },
      { title: "動画アーカイブ", href: "/videos" },
    ],
  },
];

export const Default: Story = {
  args: {
    siteMap: extensiveSiteMap,
  },
};
