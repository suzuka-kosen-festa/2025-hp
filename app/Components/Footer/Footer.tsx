import type { ReactNode } from "react";
import Box from "@mui/material/Box";
import { grey } from "@mui/material/colors";
import List from "@mui/material/List";
import ListItem from "@mui/material/ListItem";
import ListItemButton from "@mui/material/ListItemButton";
import ListItemText from "@mui/material/ListItemText";
import Typography from "@mui/material/Typography";
import { Link } from "react-router";
import { ja } from "@/locales/ja";

// サイトマップの型定義
export interface SiteMapLink {
  title: string
  href: string
}

export interface SiteMapSection {
  sectionTitle: string
  links: SiteMapLink[]
}

export interface FooterComponentProps {
  siteMap: SiteMapSection[]
}

// 個別のリンクコンポーネント
function FooterLink({ title, href }: SiteMapLink): ReactNode {
  return (
    <ListItemButton sx={{
      paddingTop: "4px",
      paddingBottom: "4px",
      height: "fit-content",
      alignSelf: "flex-start",
      minHeight: "32px",
      width: "stretch",
    }}
    >
      <ListItemText
        primary="・"
        sx={{
          minWidth: "auto",
          marginRight: "4px",
        }}
      />
      <Link
        to={href}
        style={{
          width: "100%",
          lineHeight: "1.4",
          textDecoration: "none",
          color: "white",
        }}
        onMouseEnter={(e) => {
          e.currentTarget.style.textDecoration = "underline";
          e.currentTarget.style.color = grey[400];
        }}
        onMouseLeave={(e) => {
          e.currentTarget.style.textDecoration = "none";
          e.currentTarget.style.color = "white";
        }}
      >
        {title}
      </Link>
    </ListItemButton>
  );
}

function Footer({ siteMap }: FooterComponentProps): ReactNode {
  return (
    <Box
      component="footer"
      sx={{
        display: "flex",
        backgroundImage: "url(/images/nasa_lunar.webp)",
        backgroundSize: "cover",
        backgroundPosition: "center",
        backgroundRepeat: "no-repeat",
        flexDirection: "column",
        alignItems: "center",
        justifyContent: "center",
        width: "100%",
        padding: { xs: "60px 15px 15px", sm: "80px 20px 20px", md: "100px 20px 20px" },
        borderRadius: "50% 50% 0 0",
        transform: { xs: "scaleX(1.2)", sm: "scaleX(1.3)", md: "scaleX(1.5)" },
        transformOrigin: "center top",
        position: "relative",
        zIndex: 1,
      }}
    >
      {siteMap.length > 0 && (
        <Box
          sx={{
            "display": "flex",
            "justifyContent": "center",
            "alignItems": "flex-start",
            "gap": { xs: "20px", sm: "30px", md: "40px" },
            "flexWrap": "wrap",
            "width": "100%",
            "maxWidth": "1200px",
            "transform": { xs: "scaleX(0.83)", sm: "scaleX(0.77)", md: "scaleX(0.67)" }, // 逆変換
            "transformOrigin": "center center",
            "@media (max-width: 768px)": {
              flexDirection: "column",
              alignItems: "center",
            },
          }}
        >
          {siteMap.map(section => (
            <Box
              key={`${section.sectionTitle}`}
              sx={{
                display: "flex",
                minWidth: "160px",
                width: "fit-content",
                height: "fit-content",
                alignItems: "flex-start",
                justifyContent: "flex-start",
                flexDirection: "column",
                overflow: "visible",
                padding: "0",
                color: "white",
                marginBottom: "20px",
              }}
            >
              <Typography
                sx={{
                  width: "100%",
                  fontWeight: "bold",
                  marginBottom: "8px",
                  paddingBottom: "4px",
                  borderBottom: "2px solid",
                }}
              >
                {section.sectionTitle}
              </Typography>
              <List
                sx={{
                  display: "flex",
                  width: "100%",
                  minWidth: "160px",
                  height: "fit-content",
                  flexDirection: "column",
                  alignItems: "flex-start",
                  justifyContent: "flex-start",
                  gap: "2px",
                  overflow: "visible",
                  padding: "0",
                }}
              >
                {section.links.map(link => (
                  <ListItem
                    key={`${link.title}`}
                    disablePadding
                    sx={{
                      width: "stretch",
                    }}
                  >
                    <FooterLink title={link.title} href={link.href} />
                  </ListItem>
                ))}
              </List>
            </Box>
          ))}
        </Box>
      )}
      <Box
        sx={{
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          width: "100%",
          padding: { xs: "15px 0", sm: "20px 0" },
          marginTop: { xs: "20px", sm: "25px", md: "30px" },
          color: "white",
          transform: { xs: "scaleX(0.83)", sm: "scaleX(0.77)", md: "scaleX(0.67)" }, // 逆変換
          transformOrigin: "center center",
          fontSize: { xs: "0.8rem", sm: "0.9rem", md: "1rem" },
        }}
      >
        {ja.footer.copyright}
      </Box>
    </Box>
  );
}

export default Footer;
