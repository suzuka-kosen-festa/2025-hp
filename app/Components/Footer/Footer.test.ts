// @vitest-environment jsdom

import { composeStories } from "@storybook/react";

import { expect, it } from "vitest";

import * as stories from "./Footer.stories";

const { Default } = composeStories(stories);
it("footer snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
