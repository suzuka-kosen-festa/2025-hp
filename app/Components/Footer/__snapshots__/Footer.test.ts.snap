// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`footer snapshot 1`] = `
<div>
  <footer
    class="MuiBox-root css-1sc8nln"
  >
    <div
      class="MuiBox-root css-tn7cqw"
    >
      <div
        class="MuiBox-root css-30dmb0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-14smj9a-MuiTypography-root"
        >
          ホーム
        </p>
        <ul
          class="MuiList-root MuiList-padding css-oyuda3-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/"
                style="line-height: 1.4; text-decoration: none;"
              >
                トップページ
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/news"
                style="line-height: 1.4; text-decoration: none;"
              >
                お知らせ
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/events"
                style="line-height: 1.4; text-decoration: none;"
              >
                イベント情報
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/access"
                style="line-height: 1.4; text-decoration: none;"
              >
                アクセス
              </a>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="MuiBox-root css-30dmb0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-14smj9a-MuiTypography-root"
        >
          高専祭について
        </p>
        <ul
          class="MuiList-root MuiList-padding css-oyuda3-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/about"
                style="line-height: 1.4; text-decoration: none;"
              >
                高専祭とは
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/committee"
                style="line-height: 1.4; text-decoration: none;"
              >
                実行委員会
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/history"
                style="line-height: 1.4; text-decoration: none;"
              >
                歴史
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/philosophy"
                style="line-height: 1.4; text-decoration: none;"
              >
                理念・目標
              </a>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="MuiBox-root css-30dmb0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-14smj9a-MuiTypography-root"
        >
          参加・出展
        </p>
        <ul
          class="MuiList-root MuiList-padding css-oyuda3-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/participation"
                style="line-height: 1.4; text-decoration: none;"
              >
                一般参加
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/exhibition"
                style="line-height: 1.4; text-decoration: none;"
              >
                出展申込
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/schedule"
                style="line-height: 1.4; text-decoration: none;"
              >
                スケジュール
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/map"
                style="line-height: 1.4; text-decoration: none;"
              >
                会場マップ
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/notes"
                style="line-height: 1.4; text-decoration: none;"
              >
                注意事項
              </a>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="MuiBox-root css-30dmb0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-14smj9a-MuiTypography-root"
        >
          サポート
        </p>
        <ul
          class="MuiList-root MuiList-padding css-oyuda3-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/faq"
                style="line-height: 1.4; text-decoration: none;"
              >
                よくある質問
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/contact"
                style="line-height: 1.4; text-decoration: none;"
              >
                お問い合わせ
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/privacy"
                style="line-height: 1.4; text-decoration: none;"
              >
                プライバシーポリシー
              </a>
            </div>
          </li>
        </ul>
      </div>
      <div
        class="MuiBox-root css-30dmb0"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-14smj9a-MuiTypography-root"
        >
          アーカイブ
        </p>
        <ul
          class="MuiList-root MuiList-padding css-oyuda3-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/archive"
                style="line-height: 1.4; text-decoration: none;"
              >
                過去の高専祭
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/gallery"
                style="line-height: 1.4; text-decoration: none;"
              >
                写真ギャラリー
              </a>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters css-a2xkym-MuiListItem-root"
          >
            <div
              class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1x31gb3-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root css-gfnhub-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  ・
                </span>
              </div>
              <a
                data-discover="true"
                href="/videos"
                style="line-height: 1.4; text-decoration: none;"
              >
                動画アーカイブ
              </a>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="MuiBox-root css-123s6x4"
    >
      ©2025 鈴鹿工業高等専門学校 高専祭実行委員会
    </div>
  </footer>
</div>
`;
